common/lib/xmodule/xmodule/modulestore/courseware_index.py

```diff
@@ -127,9 +130,40 @@ def remove_index_item_location(item_location):
         if raise_on_error and error_list:
             raise SearchIndexingError(_('Error(s) present during indexing'), error_list)
 
+        return indexed_count
+
+    @classmethod
+    def do_publish_index(cls, modulestore, location, delete=False, raise_on_error=False):
+        """
+        Add to courseware search index published section and children
+        """
+        indexed_count = cls.add_to_search_index(modulestore, location, delete, raise_on_error)
+        #Google Analytics - log index published section request
```

Comment:
It would be good to confirm this with @shnayder, as I believe the desire is to have these events logged for tracking.  This kind of information would also be useful to researchers, not just marketers.